iceoryx_binding_c/test/moduletests/test_chunk.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+extern "C" {
+#include "iceoryx_binding_c/chunk.h"
+#include "iceoryx_binding_c/publisher.h"
+#include "iceoryx_binding_c/runtime.h"
+}
+
+#include "testutils/roudi_gtest.hpp"
+
+namespace
+{
+using namespace iox;
+using namespace iox::mepoo;
+
+
+class BindingC_Chunk_test : public RouDi_GTest
+{
+  public:
+    void SetUp() override
+    {
+    }
+
+    void TearDown() override
+    {
+    }
+};
+
+TEST_F(BindingC_Chunk_test, GettingChunkHeaderFromNonConstUserPayloadWorks)
+{
+    iox_runtime_init("hypnotoad");
+
+    iox_pub_storage_t publisherStorage;
+    iox_pub_t publisher = iox_pub_init(&publisherStorage, "All", "Glory", "Hypnotoad", nullptr);
+
+    constexpr uint32_t USER_PAYLOAD_SIZE(42U);
+    void* userPayload;
```

Comment:
we should initialize the `userPayload = nullptr;` otherwise we may have a passing test since the random value inside of this variable just fits. (for all tests)